test/Microsoft.Extensions.Configuration.Test/ConfigurationTest.cs

```diff
@@ -333,5 +333,27 @@ public void NewTokenAfterReloadIsNotChanged()
             Assert.False(hasChanged3);
             Assert.NotSame(token1, token2);
         }
+
+        [Fact]
+        public void KeyStartingWithColonMeansFirstSectionHasEmptyName()
+        {
+            // Arrange
+            var dict = new Dictionary<string, string>
+            {
+                [":Key2"] = "value"
+            };
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dict);
+            var config = configurationBuilder.Build();
+
+            // Act
+            var children = config.GetChildren().ToArray();
+
+            // Assert
+            Assert.Equal(1, children.Length);
+            Assert.Equal(string.Empty, children.First().Key);
+            Assert.Equal(1, children.First().GetChildren().Count());
+            Assert.Equal("Key2", children.First().GetChildren().First().Key);
+        }
     }
 }
```

Comment:
I recommend either making this a theory and add more cases, or adding more Facts to cover a few more scenarios:

  - `:foo` which you already cover
  - `foo::bar`
  - `foo:` 
  - `:foo::bar` 

etc.